tools/pythonpkg/tests/fast/arrow/test_arrow_replacement_scan.py

```diff
@@ -27,9 +21,24 @@ def test_arrow_table_replacement_scan(self, duckdb_cursor):
             assert con.execute("select count(*) from userdata_parquet_table").fetchone() == (1000,)
             assert con.execute("select count(*) from df").fetchone() == (1000,)
 
+    def test_arrow_pycapsule_replacement_scan(self, duckdb_cursor):
+        tbl = pa.Table.from_pydict({'a': [1, 2, 3, 4, 5, 6, 7, 8, 9]})
+        capsule = tbl.__arrow_c_stream__()
+
+        rel = duckdb_cursor.sql("select * from capsule")
+        assert rel.fetchall() == [(i,) for i in range(1, 10)]
+
+        capsule = tbl.__arrow_c_stream__()
+        rel = duckdb_cursor.sql("select * from capsule where a > 3 and a < 5")
+        assert rel.fetchall() == [(4,)]
+
+        tbl = pa.Table.from_pydict({'a': [1, 2, 3], 'b': [4, 5, 6], 'c': [7, 8, 9], 'd': [10, 11, 12]})
+        capsule = tbl.__arrow_c_stream__()
+
+        rel = duckdb_cursor.sql("select b, d from capsule")
+        assert rel.fetchall() == [(i, i + 6) for i in range(4, 7)]
```

Comment:
We could theoretically run this test without having any pyarrow dependency, right?
Maybe we add one for that

Reply:
We could, but I don't know if there's a library that produces an arrow pycapsule that does not depend on pyarrow

Reply:
I think they do, but I thought Polars still has a dependency on pyarrow?
Might be entirely wrong on that actually, I'll check if I can install+run polars with pyarrow uninstalled

Reply:
Added the test, they do not rely on pyarrow 👍